File: programmers/42576-complete-runner.py
# https://programmers.co.kr/learn/courses/30/lessons/42576

def solution(participant, completion):
    cdict = {}    
    for p in completion: 
        if p not in cdict:
            cdict[p] = 1
        else:
            cdict[p] += 1
    
    for p in participant:
        if p not in cdict or cdict[p] == 0:
            return p
        else:
            cdict[p] -= 1
    return "honux"
           

File: programmers/42576-complete-runner.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
	string answer = "";    
	unordered_map<string, int> cmap;    
	for (auto p: completion) {
		if (cmap.find(p) != cmap.end()) {
			cmap[p]++;
		} else {
			cmap[p] = 1;
		}
	}

	for (auto p: participant) {
		if ( cmap.find(p) == cmap.end() || cmap[p] == 0) { 
			return p;
		} 
		cmap[p]--;
	}    
	//can't reach here by condtion
	return "Teemo";
}
